lib/Gedmo/Sortable/Mapping/Event/Adapter/ORM.php

```diff
@@ -39,12 +39,38 @@ private function addGroupWhere(QueryBuilder $qb, $groups)
                 $qb->andWhere($qb->expr()->isNull('n.'.$group));
             } else {
                 $qb->andWhere('n.'.$group.' = :group__'.$i);
-                $qb->setParameter('group__'.$i, $value);
+                $qb->setParameter('group__'.$i, $this->getGroupValue($value), $this->getGroupType($value));
             }
             $i++;
         }
     }
 
+    /**
+     * @param $value
+     * @return mixed
+     */
+    private function getGroupValue($value)
+    {
+        if (!is_object($value)) {
```

Comment:
is_object check is not enough, you will have to create an `isEntity` function to check that the object is a mapped entity.